nixos/modules/services/networking/netbird.nix

```diff
@@ -1,171 +1,513 @@
-{
-  config,
-  lib,
-  pkgs,
-  ...
+{ config
+, lib
+, pkgs
+, ...
 }:
 
 let
   inherit (lib)
-    attrNames
+    attrValues
+    concatLists
+    concatStringsSep
+    escapeShellArgs
+    filterAttrs
     getExe
     literalExpression
     maintainers
+    makeBinPath
     mapAttrs'
+    mapAttrsToList
     mkDefault
-    mkEnableOption
     mkIf
     mkMerge
     mkOption
+    mkOptionDefault
     mkPackageOption
+    mkRemovedOptionModule
     nameValuePair
     optional
+    optionalString
+    toShellVars
+    versionAtLeast
     versionOlder
     ;
 
   inherit (lib.types)
+    anything
     attrsOf
+    bool
+    enum
+    package
     port
     str
     submodule
     ;
 
-  kernel = config.boot.kernelPackages;
+  inherit (config.boot) kernelPackages;
+  inherit (config.boot.kernelPackages) kernel;
 
   cfg = config.services.netbird;
+
+  toClientList = fn: map fn (attrValues cfg.clients);
+  toClientAttrs = fn: mapAttrs' (_: fn) cfg.clients;
+
+  hardenedClients = filterAttrs (_: client: client.hardened) cfg.clients;
+  toHardenedClientList = fn: map fn (attrValues hardenedClients);
+  toHardenedClientAttrs = fn: mapAttrs' (_: fn) hardenedClients;
+
+  nixosConfig = config;
 in
 {
   meta.maintainers = with maintainers; [
     misuzu
     thubrecht
+    nazarewk
   ];
   meta.doc = ./netbird.md;
 
+  imports = [
+    (mkRemovedOptionModule [ "services" "netbird" "tunnels" ]
+      "The option `services.netbird.tunnels` has been renamed to `services.netbird.clients`")
+  ];
+
   options.services.netbird = {
-    enable = mkEnableOption "Netbird daemon";
+    enable = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Enables backwards compatible Netbird client service.
+
+        This is strictly equivalent to:
+
+        ```nix
+        services.netbird.clients.wt0 = {
+          port = 51820;
+          name = "netbird";
+          interface = "wt0";
+          hardened = false;
+        };
+        ```
+      '';
+    };
     package = mkPackageOption pkgs "netbird" { };
 
-    tunnels = mkOption {
+    ui.enable = mkOption {
+      type = bool;
+      default = config.services.xserver.displayManager.sessionPackages != [ ];
```

Comment:
Is there any rationale for using `config.services.xserver.displayManager.sessionPackages != [ ]` over `config.services.xserver.enable`?

If yes, note that `config.services.xserver.displayManager.sessionPackages` has been renamed to `config.services.displayManager.sessionPackages`

Reply:
yeah, I just realized the source of my month-long errors about using renamed option.
checking for `sessionPackages` makes more sense because those can also contain Wayland sessions, not just X sessions.

Reply:
On my configuration the sessionPackages are empty because I use startx and no display manager. So maybe we can use ` || config.services.xserver.enable` as an additional clause?